exchanges/okx/helpers.go

```diff
@@ -0,0 +1,214 @@
+	default:
+		return "", fmt.Errorf("%w %v", order.ErrUnsupportedOrderType, orderType)
+	}
+}
+
+// GetAssetsFromInstrumentTypeOrID parses an instrument type and instrument ID and returns a list of assets
+// that the currency pair is associated with
+func (ok *Okx) GetAssetsFromInstrumentTypeOrID(instType, instrumentID string) ([]asset.Item, error) {
+	if instType != "" {
+		a, err := AssetTypeFromInstrumentType(instType)
+		if err != nil {
+			return nil, err
+		}
+		if a != asset.Empty {
+			return []asset.Item{a}, nil
+		}
+	}
+	if instrumentID == "" {
+		return nil, errMissingInstrumentID
+	}
+	pf, err := ok.CurrencyPairs.GetFormat(asset.Spot, true)
+	if err != nil {
+		return nil, err
+	}
+	splitSymbol := strings.Split(instrumentID, pf.Delimiter)
+	if len(splitSymbol) <= 1 {
+		return nil, fmt.Errorf("%w %v", currency.ErrCurrencyNotSupported, instrumentID)
+	}
+	pair, err := currency.NewPairDelimiter(instrumentID, pf.Delimiter)
+	if err != nil {
+		return nil, err
+	}
+	switch {
+	case len(splitSymbol) == 2:
+		resp := make([]asset.Item, 0, 2)
+		enabled, err := ok.IsPairEnabled(pair, asset.Spot)
+		if err != nil {
+			return nil, err
+		}
+		if enabled {
+			resp = append(resp, asset.Spot)
+		}
+		enabled, err = ok.IsPairEnabled(pair, asset.Margin)
+		if err != nil {
+			return nil, err
+		}
+		if enabled {
+			resp = append(resp, asset.Margin)
+		}
+		if len(resp) > 0 {
+			return resp, nil
+		}
+	case len(splitSymbol) > 2:
+		switch splitSymbol[len(splitSymbol)-1] {
+		case "SWAP", "swap":
+			enabled, err := ok.IsPairEnabled(pair, asset.PerpetualSwap)
+			if err != nil {
+				return nil, err
+			}
+			if enabled {
+				return []asset.Item{asset.PerpetualSwap}, nil
+			}
+		case "C", "P", "c", "p":
+			enabled, err := ok.IsPairEnabled(pair, asset.Options)
+			if err != nil {
+				return nil, err
+			}
+			if enabled {
+				return []asset.Item{asset.Options}, nil
+			}
+		default:
+			enabled, err := ok.IsPairEnabled(pair, asset.Futures)
+			if err != nil {
+				return nil, err
+			}
+			if enabled {
+				return []asset.Item{asset.Futures}, nil
+			}
+		}
+	}
+	return nil, fmt.Errorf("%w  '%v' or currency not enabled '%v'", asset.ErrNotSupported, instType, instrumentID)
+}
+
+// AssetTypeFromInstrumentType returns an asset Item instance given and Instrument Type string
+func AssetTypeFromInstrumentType(instrumentType string) (asset.Item, error) {
+	switch strings.ToUpper(instrumentType) {
+	case instTypeSwap, instTypeContract:
+		return asset.PerpetualSwap, nil
+	case instTypeSpot:
+		return asset.Spot, nil
+	case instTypeMargin:
+		return asset.Margin, nil
+	case instTypeFutures:
+		return asset.Futures, nil
+	case instTypeOption:
+		return asset.Options, nil
+	case "":
+		return asset.Empty, nil
+	default:
+		return asset.Empty, asset.ErrNotSupported
+	}
+}
+
+func (ok *Okx) validatePlaceOrderParams(arg *PlaceOrderRequestParam) error {
+	if *arg == (PlaceOrderRequestParam{}) {
+		return common.ErrEmptyParams
+	}
```

Comment:
I don't think we need this check, because if it's empty, the next if statement catches it, and is the best representation of the problem: All trades need at least an instrumentID